nilearn/datasets/_utils.py

```diff
@@ -159,13 +154,12 @@ def _chunk_read_(
             "Warning: total size could not be determined.",
             verbose=verbose,
             msg_level=2,
-            stack_level=2,
         )
         logger.log(
-            f"Full stack trace: {e}",
+            "Full stack trace: %s",
```

Comment:
I'm a bit puzzled by this and following changes ?

Reply:
That's all on purpose:
* String interpolation is "usually" deferred to the log function, so that interpolation occurs only when the message will be printed according the log level (lazy interpolation). See [G004](https://docs.astral.sh/ruff/rules/logging-f-string/). The efficiency of this rule is debatable though (and has been debated).
* The `stack_level` argument is already specified by `logger.log` in this case. Repeating it here, with a different value, results in warnings.

Reply:
With that said, I can defer these changes to a different PR, especially the f-string removal.

Reply:
no problem: your assumption was a sound one !